homeassistant/components/devolo_home_control/switch.py

```diff
@@ -0,0 +1,161 @@
+"""Platform for light integration."""
+import logging
+
+from devolo_home_control_api.homecontrol import get_sub_device_uid_from_element_uid
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DOMAIN
+
+logging.basicConfig(level=logging.DEBUG)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Get all devices and setup the switch devices via config entry."""
+    devices = hass.data[DOMAIN]["homecontrol"].binary_switch_devices
+
+    devices_list = []
+    for device in devices:
+        for i in range(len(device.binary_switch_property)):
+            devices_list.append(
+                DevoloSwitch(
+                    hass=hass,
+                    device_instance=device,
+                    sub_uid=get_sub_device_uid_from_element_uid(
+                        [*device.binary_switch_property][i]
+                    ),
+                )
+            )
+    async_add_entities(devices_list, False)
+
+
+class DevoloSwitch(SwitchDevice):
+    """Representation of an Awesome Light."""
+
+    def __init__(self, hass, device_instance, sub_uid):
+        """Initialize an devolo Switch."""
+        self._device_instance = device_instance
+
+        # Create the unique ID
+        if sub_uid is not None:
+            self._unique_id = self._device_instance.uid + "#" + str(sub_uid)
+        else:
+            self._unique_id = self._device_instance.uid
+
+        self._homecontrol = hass.data[DOMAIN]["homecontrol"]
+        self._name = self._device_instance.itemName
+        self._available = self._device_instance.is_online()
+
+        # Get the brand and model information
+        try:
+            self._brand = self._device_instance.brand
+            self._model = self._device_instance.name
+        except AttributeError:
+            self._brand = None
+            self._model = None
+
+        self._binary_switch_property = self._device_instance.binary_switch_property.get(
+            "devolo.BinarySwitch:" + self._unique_id
+        )
+        self._is_on = self._binary_switch_property.state
+
+        if hasattr(self._device_instance, "consumption_property"):
+            self._consumption = self._device_instance.consumption_property.get(
+                "devolo.Meter:" + self._unique_id
+            ).current
+        else:
+            self._consumption = None
+        self._subscriber = Subscriber(self._device_instance.itemName, device=self)
```

Comment:
Don't pass the entity to the subscriber. Use our dispatch helper to signal the entity to update from the subscriber. The entity api should not be used outside the entity since the api is only valid after the entity has been added to home assistant.

See eg the tuya integration for how to use our dispatch helper.

https://github.com/home-assistant/core/blob/dd4a350bd585ea17ef799670cc11b2017b1de110/homeassistant/components/tuya/__init__.py#L97-L99

https://github.com/home-assistant/core/blob/dd4a350bd585ea17ef799670cc11b2017b1de110/homeassistant/components/tuya/__init__.py#L122-L126

Reply:
Use own subscriber, but don't pass the whole entity in init